### PR TITLE
Issues/95 - Add Run 2.1i DR1 Object Table tutorial

### PR DESCRIPTION
**Pull request by wmwv**
Adds a tutorial stepping through the Run 2.1i DR1 Object Table.

### CODE REVIEW

File: validation/validate_dc2_run2.1i_object_table.ipynb [hunk 1]
```diff
@@ -0,0 +1,713 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for filt in filters:\n",
+    "    df['e_{}'.format(filt)], df['e1_{}'.format(filt)], df['e2_{}'.format(filt)] = \\\n",
```
Comment (joezuntz) [inline validation/validate_dc2_run2.1i_object_table.ipynb]: When I get to this cell on jupyter.nersc.gov the kernel silently dies.  I think it's running out of memory.

So I can't really review beyond here.

File: validation/validate_dc2_run2.1i_object_table.ipynb [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "DC2 Run 2.x WFD and DDF regions\n",
+    "https://docs.google.com/document/d/18nNVImxGioQ3tcLFMRr67G_jpOzCIOdar9bjqChueQg/view\n",
+    "https://github.com/LSSTDESC/DC2_visitList/blob/master/DC2visitGen/notebooks/DC2_Run2_regionCoords_WFD.ipynb\n",
+    "\n",
+    "| Location        | RA (degrees) | Dec (degrees) | RA (degrees) | Dec (degrees) |\n",
```
[the remark this answers is not in the document]
Comment (wmwv) [reply validation/validate_dc2_run2.1i_object_table.ipynb]: I fixed this by starting each table row with a "|".  I previously only had a few rows starting with a "|" and the rest didn't.  The GitHub viewer apparently didn't like that.  Thanks for catching.

File: validation/validate_dc2_run2.1i_object_table.ipynb [hunk 3]
```diff
@@ -0,0 +1,713 @@
...
+    "    ax.add_patch(ddf_region)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_ra_dec(df)"
```
[the remark this answers is not in the document]
Comment (wmwv) [reply validation/validate_dc2_run2.1i_object_table.ipynb]: The goal is for run Notebooks to be available in the `rendered` branch.
I will make sure that this works and exists in that branch.

File: validation/validate_dc2_run2.1i_object_table.ipynb [hunk 4]
```diff
@@ -0,0 +1,713 @@
...
+    "    # This is a slightly tedious way of defining a symmetric trapezoid\n",
+    "    # Could consider using geopandas, but that adds dependency\n",
+    "    dec_size = corners[1][1] - corners[2][1] # deg\n",
+    "\n",
+    "    ra_left_side_delta = corners[1][0] - corners[2][0]\n",
+    "    ra_right_side_delta = corners[0][0] - corners[3][0]\n",
+    "    ra_left_side_slope = ra_left_side_delta / dec_size\n",
+    "    ra_right_side_slope = ra_right_side_delta / dec_size\n",
+    "\n",
+    "    inside_ra = (corners[2][0] + ra_left_side_slope * (df['dec'] - corners[2][1]) < df['ra']) & \\\n",
```
[the remark this answers is not in the document]
Comment (wmwv) [reply validation/validate_dc2_run2.1i_object_table.ipynb]: Oh, thanks!  Clearly a residual from promoting to a function without actually changing the variables.

File: validation/validate_dc2_run2.1i_object_table.ipynb (hunk 1 above)
Comment (wmwv) [reply validation/validate_dc2_run2.1i_object_table.ipynb]: Can you try with the restricted one-tract catalog?
I've added it as an option in the first cell to allow someone to run the Notebook in a finite amount of time.

I will also add a completed run to the `rendered` branch.

File: validation/validate_dc2_run2.1i_object_table.ipynb [hunk 5]
```diff
@@ -0,0 +1,733 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The overall object density distribution looks good.\n",
+    "\n",
+    "Notes:\n",
+    "* We're missing the DDF region, which was specifically not included in this DR1 coadd.\n",
+    "* We're missing a bit of the South-West (lower-left, \"North Up, East Left\") corner.\n",
+    "* There are also a few patches that failed within the main region.\n",
+    "* There is an overall gradient N/S in object density, because we're plotting in rectilinear RA, Dec bins, which means that bins at the bottom in RA cover less area than those at the top.\n",
```
Comment (sschmidt23) [inline validation/validate_dc2_run2.1i_object_table.ipynb]: If you plotted only stars/extendedness==0 would you see a gradient due to Galactic latitude, or is this too small a range to spot that?  If it is visible, might be a nice additional plot to make

File: validation/validate_dc2_run2.1i_object_table.ipynb [hunk 6]
```diff
@@ -0,0 +1,733 @@
...
+    "plt.ylabel('blendedness');"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.hexbin(good['mag_i'], good['extendedness'],\n",
```
Comment (sschmidt23) [inline validation/validate_dc2_run2.1i_object_table.ipynb]: The plot shows that extendedness is a binary quantity, is it worth it to also show mag_psf - mag_cModel vs mag_cModel (i.e. Figure 16 from the Bosch et al HSC paper) to remind people that extendedness, while binary, is sensitive to an ad hoc threshold choice in the delta_mag?  That might be more informative when paired with the binary extendedness plot.

**Comment by RobertLuptonTheGood** [comment]
We (== DM) have SVM (?) code to do a better job of s/g separation.  It's been used trained on HSC data (using HST COSMOS imaging as space truth, although HSC is too deep), and we could probably run in on DESC sims too.

			R

File: validation/validate_dc2_run2.1i_object_table.ipynb (hunk 6 above)
Comment (wmwv) [reply validation/validate_dc2_run2.1i_object_table.ipynb]: Good suggestions.  Done.

File: validation/validate_dc2_run2.1i_object_table.ipynb [hunk 7]
```diff
@@ -0,0 +1,713 @@
...
+    "    ellipticity(df['Ixx_{}'.format(filt)], df['Ixy_{}'.format(filt)], df['Iyy_{}'.format(filt)])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def inside_trapezoid(corners, ra, dec):\n",
```
[the remark this answers is not in the document]
Comment (wmwv) [reply validation/validate_dc2_run2.1i_object_table.ipynb]: ⚪️ ▪️

**Comment by wmwv** [comment]
Updates now pushed:
* Default to `*_tract4850`.  This takes ~10 seconds to load.
* Added subsampling logic to make easy subsamples when loading full DC2.  Scanning the full set of data files still takes ~5 minutes, but the reduced memory footprint means the JupyterHub node doesn't spend all of its time swapping.
* Added Mandelbaum N(mag_i) HSC comparison for galaxies.

File: validation/validate_dc2_run2.1i_object_table.ipynb (hunk 5 above)
Comment (wmwv) [reply validation/validate_dc2_run2.1i_object_table.ipynb]: No, too small a range at too high above the Galactic plane to see a trend.

**Comment by wmwv** [comment]
Thanks @joezuntz  @sschmidt23 for your reviews.

Even in the absence of formal approval from the reviewers, I'm taking the liberty of merging this to get this Notebook in front of the wider collaboration.

**Pull request merged**